src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -312,6 +313,20 @@ private string FinishFixingName(string name)
             if (!string.IsNullOrEmpty(WordSeparator))
             {
                 words = name.Split(new[] { WordSeparator }, StringSplitOptions.RemoveEmptyEntries);
+
+                if (CapitalizationScheme == Capitalization.AllUpper || CapitalizationScheme == Capitalization.AllLower)
+                {
+                    if (words.Count() == 1) // Only Split if words have not been split before 
+                    {
+                        var parts = StringBreaker.GetParts(name, true);
```

Comment:
For Booleans like this, we'll add a named argument (`foo:`) so it's easier to read.

Reply:
Done